### PR TITLE
Fn cve search

### PR DESCRIPTION
**Pull request by neetinkandhare**

### CODE REVIEW

File: fn_cve_search/fn_cve_search/LICENSE [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Copyright © IBM Corporation 2010, 2018
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/LICENSE]:
```
Copyright © IBM Corporation 2010, 2019
```

File: fn_cve_search/fn_cve_search/components/function_cve_browse.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+# (c) Copyright IBM Corp. 2010, 2018. All Rights Reserved.
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_browse.py]: 2019

File: fn_cve_search/fn_cve_search/components/function_cve_search.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
+# (c) Copyright IBM Corp. 2010, 2018. All Rights Reserved.
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_search.py]: 2019

File: fn_cve_search/fn_cve_search/util/config.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# (c) Copyright IBM Corp. 2010, 2018. All Rights Reserved.
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/util/config.py]: 2019

File: fn_cve_search/README.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+#About This Package
```
Comment (sjcurtinibm) [inline fn_cve_search/README.md]: Is it possible to refactor these docs similarly to: https://github.com/neetinkandhare/resilient-community-apps/blob/alien-vault/fn_alienvault_otx/README.md

File: fn_cve_search/fn_cve_search/components/function_cve_browse.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+            # Variables to Store Parsed CVE API Data
+            result_data_dict = dict()
+
+            log = logging.getLogger(__name__)
+            log.info("cve_browse_data: %s", cve_browse_data)
+            log.info("cve_browse_criteria: %s", cve_browse_criteria)
+            log.info("cve_vendor: %s", cve_vendor)
+
+            yield StatusMessage("starting...")
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_browse.py]: Something like: "Getting list of Vendors from the CVE Database"

File: fn_cve_search/fn_cve_search/components/function_cve_browse.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+            if api_call_type == 'browse':
+                result_data_dict['api_call'] = 'browse'
+                _browse_dict_list = self._parse_browse_results(api_data=browse_data_content)
+                result_data_dict['content'].extend(_browse_dict_list)
+            elif api_call_type == 'db':
+                result_data_dict['api_call'] = 'db'
+                result_data_dict['content'].append(browse_data_content)
+
+            log.debug("The Data Received from CVE: {}".format(result_data_dict))
+            yield StatusMessage("done...")
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_browse.py]: A better message like: "List of Vendors received from CVE Database"

File: fn_cve_search/fn_cve_search/components/function_cve_search.py [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+                        else:
+                            """
+                            If 'max_res_counter' Counter Value reaches to '0' data should not be accumulated.
+                            """
+                            pass
+                    else:
+                        """
+                        If given Published Date is not in range data should not be accumulated.
+                        """
+                        pass
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_search.py]: No need for this else condition if all we are doing is `passing`

File: fn_cve_search/fn_cve_search/components/function_cve_search.py [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+                    if (search_pub_date_timestamp >= cve_pub_date_from) and (
+                            search_pub_date_timestamp <= cve_pub_date_to):
+                        if max_res_counter != 0:
+                            tmp_search_data.append(search_data_dict)
+                            max_res_counter -= 1
+                        else:
+                            """
+                            If 'max_res_counter' Counter Value reaches to '0' data should not be accumulated.
+                            """
+                            pass
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_search.py]: No need for this else condition if all we are doing is `passing`

File: fn_cve_search/fn_cve_search/components/function_cve_search.py [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+                elif cve_pub_date_from is not None and cve_pub_date_to is None:
+                    if search_pub_date_timestamp >= cve_pub_date_from:
+                        if max_res_counter != 0:
+                            tmp_search_data.append(search_data_dict)
+                            max_res_counter -= 1
+                        else:
+                            """
+                            If 'max_res_counter' Counter Value reaches to '0' data should not be accumulated.
+                            """
+                            pass
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_search.py]: Again, remove all these `pass`s

File: fn_cve_search/fn_cve_search/components/function_cve_search.py [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+
+            # Getting Max Result Display Flag from app.config
+            MAX_RESULTS_RETURN = int(self.options.get('max_results_display'))
+
+            # Variables to Store Parsed CVE API Data
+            _result_search_data = dict()
+
+            log = logging.getLogger(__name__)
+            log.info("cve_search_data: %s", cve_search_data)
+            log.info("cve_search_criteria: %s", cve_search_criteria)
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_search.py]: What is `cve_search_criteria`?

Can we give an example of all Function Inputs and an explanation of each in our README doc?

File: fn_cve_search/fn_cve_search/components/function_cve_search.py [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+            log.info("cve_id: %s", cve_id)
+            log.info("cve_vendor: %s", cve_vendor)
+            log.info("cve_product: %s", cve_product)
+            log.info("cve_published_date_from: %s", cve_published_date_from)
+            log.info("cve_published_date_to: %s", cve_published_date_to)
+
+            yield StatusMessage("starting...")
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_search.py]: A better message like: "Searching the CVE Database. ID: {0}, Vendor: {1}......etc

File: fn_cve_search/fn_cve_search/components/function_cve_search.py [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+                elif api_call_type == 'cve':
+                    _result_search_data['api_call'] = 'cve'
+                    cve_data_list = self._parse_cve_results(_browse_data_content)
+                    _result_search_data['content'].extend(cve_data_list)
+                elif api_call_type == 'last':
+                    _result_search_data['api_call'] = 'last'
+                    last_data_list = self._parse_last_cve_results(_browse_data_content, MAX_RESULTS_RETURN)
+                    _result_search_data['content'].extend(last_data_list)
+            log.debug("The Data Received from CVE DB : {}".format(_result_search_data))
+            yield StatusMessage("done...")
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/components/function_cve_search.py]: "Successfully searched the CVE Database"

File: fn_cve_search/tox.ini [hunk 14]
```diff
@@ -0,0 +1,17 @@
+# Tox (https://tox.readthedocs.io/) is a tool for running tests
```
Comment (sjcurtinibm) [inline fn_cve_search/tox.ini]: This file is not needed as there are no unit tests...

File: fn_cve_search/fn_cve_search/util/cve.py [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+    response_json_object = dict()
+    try:
+        response_object = requests.get(url=rest_url)
+        response_code = response_object.status_code
+        if response_code == 200:
+            response_json_object['content'] = response_object.json()
+            response_json_object['api_call'] = api_call
+            return response_json_object
+        else:
+            raise FunctionError("CVE API Call Failed with status code : {}".format(response_code))
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/util/cve.py]: There have been issues when using `FunctionError` in a nested `try` block. Better to use `ValueError` here

File: fn_cve_search/fn_cve_search/util/cve.py [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+    except Exception as call_err:
+        raise FunctionError("CVE API Call Failed : {}".format(call_err))
```
Comment (sjcurtinibm) [inline fn_cve_search/fn_cve_search/util/cve.py]: Use `ValueError` here too

File: fn_cve_search/fn_cve_search/LICENSE (hunk 1 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/LICENSE]: ok

File: fn_cve_search/fn_cve_search/components/function_cve_search.py (hunk 3 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/components/function_cve_search.py]: ok

File: fn_cve_search/README.md (hunk 5 above)
Comment (neetinkandhare) [reply fn_cve_search/README.md]: yes

File: fn_cve_search/fn_cve_search/components/function_cve_browse.py (hunk 6 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/components/function_cve_browse.py]: ok

File: fn_cve_search/fn_cve_search/components/function_cve_browse.py (hunk 7 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/components/function_cve_browse.py]: ok

File: fn_cve_search/fn_cve_search/components/function_cve_search.py (hunk 11 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/components/function_cve_search.py]: we have explained about each of search criteria in the README file.

File: fn_cve_search/fn_cve_search/components/function_cve_search.py (hunk 12 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/components/function_cve_search.py]: ok

File: fn_cve_search/fn_cve_search/components/function_cve_search.py (hunk 13 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/components/function_cve_search.py]: ok

File: fn_cve_search/tox.ini (hunk 14 above)
Comment (neetinkandhare) [reply fn_cve_search/tox.ini]: file removed.

File: fn_cve_search/fn_cve_search/util/cve.py (hunk 15 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/util/cve.py]: ok

File: fn_cve_search/fn_cve_search/util/cve.py (hunk 16 above)
Comment (neetinkandhare) [reply fn_cve_search/fn_cve_search/util/cve.py]: ok

File: fn_cve_search/MANIFEST.in [hunk 17]
```diff
@@ -0,0 +1,4 @@
+include README.md
+include fn_cve_search/util/*
+include fn_cve_search/LICENSE
+include fn_cve_search/doc/*
```
Comment (sjcurtinibm) [inline fn_cve_search/MANIFEST.in]: * add `include screenshots/*`
* remove `include fn_cve_search/doc/*`

File: fn_cve_search/README.md [hunk 18]
```diff
@@ -0,0 +1,297 @@
+# CVE Search Function for IBM Resilient
```
Comment (sjcurtinibm) [inline fn_cve_search/README.md]: Nice README!

File: fn_cve_search/MANIFEST.in (hunk 17 above)
Comment (neetinkandhare) [reply fn_cve_search/MANIFEST.in]: ok

**Comment by sjcurtinibm** [comment]
Closing this PR as being handled internally as INT-758

**Pull request closed without merging**